src/vec_bcd_ppc.h

```
@@ -886,6 +1476,50 @@ vec_BCD2DFP (vBCD_t val)
 #endif
 }
 
+/** \brief Convert vector unsigned doubleword binary values to
+ * Vector unsigned 16-digit BCD values
+ *
+ * Convert a vector of 2 unsigned long int doubleword  to 2 16-digit
+ * unsigned BCD doublewords.
+ * Input doublewords should each be in the range
+ * 0-9999999999999999.
+ *
+ *
+ * |processor|Latency|Throughput|
+ * |--------:|:-----:|:---------|
+ * |power8   |  69   |1/19 cycle|
+ * |power9   |  58   |1/21 cycle|
+ *
+ * @param val a vector unsigned long int.
+ * @return a 128-bit vector treated a 2 unsigned BCD 16 digit values.
+ */
+static inline vBCD_t
+vec_BIN2BCD (vui64_t val)
+{
+#ifdef _ARCH_PWR6
+  vBCD_t t;
+  _Decimal128 x, y, z;
+  // unpack the vector into a FPRp
+  z = vec_unpack_Decimal128 ((vf64_t) val);
+  // Convert 2 long int values into 2 _Decimal64 values
+  // Then convert each _Decimal64 value into 16-digit BCD
+  __asm__(
+      "dcffix %1,%2;\n"
+      "dcffix %L1,%L2;\n"
+      "ddedpd 0,%0,%1;\n"
+      "ddedpd 0,%L0,%L1;\n"
+      : "=d" (x),
+      "=&d" (y)
+      : "d" (z)
+      : );
+  // Pack the FPRp back into a vector
+  t = (vBCD_t) vec_pack_Decimal128 (x);
+  return (t);
+#else
+  // todo no solution before P6
```

should we throw an error here?

---

No that just pisses people off.